File: worker/src/main/java/ru/variantum/worker/controller/StatsController.java
package ru.variantum.worker.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import ru.variantum.worker.domain.AnalysisEventRecord;
import ru.variantum.worker.domain.ExportEventRecord;
import ru.variantum.worker.repository.AnalysisEventRepository;
import ru.variantum.worker.repository.ExportEventRepository;

import java.util.List;

/** Показывает, что worker реально принимает и хранит события — для проверки и демо. */
@RestController
@RequestMapping("/stats")
@RequiredArgsConstructor
public class StatsController {

    private final ExportEventRepository exportEventRepository;
    private final AnalysisEventRepository analysisEventRepository;

    @GetMapping("/exports")
    public StatsResponse<ExportEventRecord> exports() {
        return new StatsResponse<>(exportEventRepository.count(), exportEventRepository.findTop20ByOrderByReceivedAtDesc());
    }

    @GetMapping("/analyses")
    public StatsResponse<AnalysisEventRecord> analyses() {
        return new StatsResponse<>(analysisEventRepository.count(), analysisEventRepository.findTop20ByOrderByReceivedAtDesc());
    }

    public record StatsResponse<T>(long total, List<T> recent) {}
}

File: worker/src/main/kotlin/ru/variantum/worker/controller/StatsController.kt
package ru.variantum.worker.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.variantum.worker.domain.AnalysisEventRecord
import ru.variantum.worker.domain.ExportEventRecord
import ru.variantum.worker.repository.AnalysisEventRepository
import ru.variantum.worker.repository.ExportEventRepository

/** Показывает, что worker реально принимает и хранит события — для проверки и демо. */
@RestController
@RequestMapping("/stats")
class StatsController(
    private val exportEventRepository: ExportEventRepository,
    private val analysisEventRepository: AnalysisEventRepository,
) {

    @GetMapping("/exports")
    fun exports(): StatsResponse<ExportEventRecord> =
        StatsResponse(exportEventRepository.count(), exportEventRepository.findTop20ByOrderByReceivedAtDesc())

    @GetMapping("/analyses")
    fun analyses(): StatsResponse<AnalysisEventRecord> =
        StatsResponse(analysisEventRepository.count(), analysisEventRepository.findTop20ByOrderByReceivedAtDesc())

    data class StatsResponse<T>(val total: Long, val recent: List<T>)
}
